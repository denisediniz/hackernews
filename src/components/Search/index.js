import React, { Component } from 'react';
import PropTypes from 'prop-types';
import Button from '../Button';

import styles from './index.module.scss';

class Search extends Component {
  componentDidMount() {
    this.input && this.input.focus();
  }

  render() {
    const { value, onChange, onSubmit, children } = this.props;

    return (
      <form onSubmit={onSubmit}>
        <div
          className={`${styles.field} ${styles['has-addons']} ${styles['has-addons-centered']}`}
        >
          <div className={styles.control}>
            <input
              className={styles.input}
              type="text"
              value={value}
              onChange={onChange}
              ref={element => (this.input = element)}
            />
          </div>
          <div className={styles.control}>
            <Button
              onClick={() => {}}
              className={`${styles.button} ${styles['is-primary']}`}
              type="submit"
            >
              {children}
            </Button>
          </div>
        </div>
      </form>
    );
  }
}

Search.propTypes = {
  value: PropTypes.node.isRequired,
  onChange: PropTypes.func.isRequired,
  onSubmit: PropTypes.func.isRequired,
  children: PropTypes.node.isRequired
};

export default Search;
