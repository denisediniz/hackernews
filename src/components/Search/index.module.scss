@import 'bulma/sass/utilities/_all.sass';
@import 'bulma/sass/form/_all.sass';
@import 'bulma/sass/elements/button.sass';
