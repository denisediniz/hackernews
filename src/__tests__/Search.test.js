import React from 'react';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';
import Search from '../components/Search';

describe('Search', () => {
  const props = {
    value: 'redux',
    onChange: () => {},
    onSubmit: () => {},
    children: 'redux'
  };

  it('renders without crashing', () => {
    shallow(<Search {...props}>Search</Search>);
  });

  it('has a valid snapshot', () => {
    const component = renderer.create(<Search {...props}>Search</Search>);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
