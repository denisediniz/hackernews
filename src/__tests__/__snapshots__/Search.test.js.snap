// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Search has a valid snapshot 1`] = `
<form
  onSubmit={[Function]}
>
  <div
    className="field has-addons has-addons-centered"
  >
    <div
      className="control"
    >
      <input
        className="input"
        onChange={[Function]}
        type="text"
        value="redux"
      />
    </div>
    <div
      className="control"
    >
      <button
        className="button is-primary"
        onClick={[Function]}
        type="submit"
      >
        Search
      </button>
    </div>
  </div>
</form>
`;
